### PR TITLE
Add support for Nodes exclusion

### PR DESCRIPTION
**Pull request by AlexNguetcha**
This pull request adds support for excluding nodes in the processing. Users can now specify a list of nodes to exclude during the `handleClickOutside event listener`, which can be useful for filtering out unwanted nodes as event target.

Here is an example :

```jsx
const ref1 = React.useRef<HTMLDivElement | null>(null);
const ref2 = React.useRef<HTMLDivElement | null>(null);
const result = useClickOutsideListener<HTMLDivElement>({...options, exclude : [ref1, ref2]});

return (
    <div>
        {/* render other elements */}
        <div ref={ref1}></div>
        <div ref={ref2}></div>
        <div id="outside-3"></div>
    </div>
)
```

`click event (or supported options events)` on  both ref1, ref2 elements will have not effect !

### CODE REVIEW

File: tests/useClickOutsideListener.test.tsx [hunk 1]
```diff
@@ -1,15 +1,19 @@
+import { fireEvent, render, waitFor, screen } from '@testing-library/react';
 import { renderHook, act } from '@testing-library/react-hooks';
-import { fireEvent } from '@testing-library/react';
 import { useClickOutsideListener, UseClickOutsideListenerOptions } from '../src';
+import { MutableRefObject } from 'react';
+import React from 'react';
```
Comment (PatrissolJuns) [inline tests/useClickOutsideListener.test.tsx]: Can please merge this 2 react import lines ?

**Review by PatrissolJuns — COMMENTED** [review]
Good job, That's a nice feature

File: tests/useClickOutsideListener.test.tsx (hunk 1 above)
Comment (AlexNguetcha) [reply tests/useClickOutsideListener.test.tsx]: Yes , of course

**Comment by AlexNguetcha** [comment]
It's done !

File: tests/useClickOutsideListener.test.tsx (hunk 1 above)
Comment (AlexNguetcha) [reply tests/useClickOutsideListener.test.tsx]: It's done !

**Pull request merged**